content/en/blog/_posts/2023-01-17-confidential-kubernetes.md

```diff
@@ -0,0 +1,324 @@
+to persistent storage to maintain confidentiality and integrity guarantees. The access to such
+sealed data needs to be well-defined. In most cases, the unsealing is bound to a TEE's identity.
+Hence, making sure the recovery can only happen in the same confidential context.
+
+This does not have to limit the flexibility of the overall system. The
+[migration agent (MA) of AMD SEV-SNP](https://www.amd.com/system/files/TechDocs/SEV-SNP-strengthening-vm-isolation-with-integrity-protection-and-more.pdf)
+allows users to migrate a confidential virtual machine to a different host system
+while keeping the security properties of the TEE intact.
+
+## Feature comparison
+
+In the following, we will dive a little bit deeper into the specific implementations,
+compare supported features and analyze their security properties.
+
+### AMD SEV
+
+AMD's [Secure Encrypted Virtualization (SEV)](https://developer.amd.com/sev/) technologies
+are a set of features to enhance the security of virtual machines on AMD's server CPUs. SEV
+transparently encrypts the memory of each VM with a unique key. SEV can also calculate a
+signature of the memory contents, which can be sent to the VM's owner as an attestation that
+the memory was encrypted correctly by the firmware.
+
+The second generation of SEV, known as
+[Encrypted State](https://www.amd.com/system/files/TechDocs/Protecting%20VM%20Register%20State%20with%20SEV-ES.pdf)
+or SEV-ES provides additional protection from the hypervisor by encrypting all CPU register contents when a VM stops running.
+
+The third generation of SEV,
+[Secure Nested Paging](https://www.amd.com/system/files/TechDocs/SEV-SNP-strengthening-vm-isolation-with-integrity-protection-and-more.pdf)
+or SEV-SNP is designed to prevent software-based integrity attacks and reduce the risk associated with
+compromised memory integrity. The basic principle of SEV-SNP integrity is that if a VM can read
+a private (encrypted) memory page, it must always read the value it last wrote.
+
+Additionally, by allowing the guest to obtain remote attestation statements dynamically,
+SNP enhances the remote attestation capabilities of SEV.
+
+AMD SEV has been implemented incrementally. New features and improvements have been added with
+each new CPU generation. The Linux community makes these features available as part of the KVM hypervisor,
+and for host and guest kernels. The first SEV features were discussed and implemented in 2016 - see
+[AMD x86 Memory Encryption Technologies](https://www.usenix.org/conference/usenixsecurity16/technical-sessions/presentation/kaplan)
+from the 2016 Usenix Security Symposium.
+
+After that, the SEV-ES feature set was implemented in
+[2020](https://www.phoronix.com/news/AMD-SEV-ES-Linux-2020-Patches). Recently, the latest feature set
+was [merged in Linux 5.19 in July 2022](https://www.phoronix.com/news/AMD-SEV-SNP-Arrives-Linux-5.19)
+and enabled support for SEV-SNP.
```

Comment:
This is only the guest support. The host support is still pending. Honestly I don't think these sections on the timeline of the Linux implementation are necessary. They are also a bit redundant with the previous description of each feature.

Reply:
shortened this section a bit